e2etest/pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// ValidateOnPrometheus checks the existance of the given metric directly on prometheus pods.
+func ValidateOnPrometheus(exec executor.Executor, query string, expected CheckType) error {
+	fmt.Println("Checking ", query, expected)
+	url := fmt.Sprintf("localhost:9090/api/v1/query?%s", (url.Values{"query": []string{query}}).Encode())
+	stdout, err := exec.Exec("wget", "-qO-", url)
```

Comment:
since this heavily relies on the exec being a prometheus pod, would it make sense having a harder restriction rather than accepting an exec.Executor?

Reply:
I'll change to accept a pod and will craft the executor out of the pod